src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -329,6 +329,34 @@ bool ClusterResourceScheduler::GetNodeResources(int64_t node_id,
 
 int64_t ClusterResourceScheduler::NumNodes() { return nodes_.size(); }
 
+void ClusterResourceScheduler::AddLocalResource(
+                                                      const std::string &resource_name,
+                                                      double resource_total) {
+  string_to_int_map_.Insert(resource_name);
+  int64_t resource_id = string_to_int_map_.Get(resource_name);
+
+  if (local_resources_.custom_resources.find(resource_id) != local_resources_.custom_resources.end()) {
+    FixedPoint total(resource_total);
+    auto &instances = local_resources_.custom_resources[resource_id];
+    instances.total[0] += total;
+    instances.available[0] += total;
+    auto &capacity = nodes_[local_node_id_].custom_resources[resource_id];
+    capacity.available += total;
+    capacity.total += total;
+  } else {
+    ResourceInstanceCapacities capacity;
+    capacity.total.resize(1);
+    capacity.total[0] = resource_total;
+    capacity.available.resize(1);
+    capacity.available[0] = resource_total;
+    local_resources_.custom_resources[resource_id] = capacity;
+    std::string client_id_string = string_to_int_map_.Get(local_node_id_);
```

Comment:
This will have teh same issue that it won't return string if `local_node_id_ ` didn't exist right? Can you either add RAY_CHECK or change the behavior of this map to return string always?

Reply:
```suggestion
    std::string node_id_string = string_to_int_map_.Get(local_node_id_);
```